src/distributor.ts

```diff
@@ -54,83 +56,111 @@ export class Distributor {
-      fromBlock: this.lastProcessedBlock ? this.lastProcessedBlock + 1 : blockNumber,
-      toBlock: blockNumber,
+      fromBlock,
+      toBlock,
       logger: this.logger,
     });
-    const block = await this.provider.getBlock(blockNumber);
+    const block = await this.provider.getBlock(toBlock);
     const assetPrice = parseFloat(utils.formatUnits((await this.market.assetPrice()).price));
 
+    this.logger.info('Distributing to keepers', {
+      args: { fromBlock, toBlock, events: events.length, assetPrice },
+    });
+
     await this.updateKeeperIndexes(events, block, assetPrice);
     await this.executeKeepers();
   }
 
-  async startProcessNewBlockConsumer() {
+  private async startProcessNewBlockConsumer() {
     // The L2 node is constantly mining blocks, one block per transaction. When a new block is received, we queue it
     // for processing in a FIFO queue. `processNewBlock` will scan its events, rebuild the index, and then run any
     // keeper tasks that need running that aren't already active.
     while (1) {
+      this.metrics.gauge(Metric.DISTRIBUTOR_QUEUE_SIZE, this.blockQueue.length);
+
       if (!this.blockQueue.length) {
         await this.delay(this.MAX_CONSUME_WAIT_TIME);
         continue;
       }
 
-      // sort in case it's unsorted for some reason
+      // Sort in case it's unsorted for some reason
       this.blockQueue.sort();
       const blockNumber = this.blockQueue.shift();
       if (blockNumber) {
+        this.logger.info('Found block in blockQueue!', {
+          args: {
+            blockNumber,
+            lastProcessedBlock: this.lastProcessedBlock,
+            remaining: this.blockQueue.length,
+          },
+        });
         await this.disburseToKeepers(blockNumber);
         this.lastProcessedBlock = blockNumber;
       }
     }
   }
 
+  // TODO: Each keeper should have a .healthcheck call which in-essence does the same thing.
+  //
+  // The metric namespace can be further chunked by keeper type e.g. PerpsV2MainnetOvm/Liquidations/KeeperUpTime
   async healthcheck(): Promise<void> {
-    this.logger.info('Performing keeper healthcheck');
-    await Promise.all([
-      this.metrics.time(Metric.KEEPER_UPTIME, Date.now() - this.START_TIME),
-      this.metrics.send(
-        Metric.KEEPER_ETH_BALANCE,
-        wei(await this.provider.getBalance(this.signer.address)).toNumber()
-      ),
-    ]);
+    const uptime = Date.now() - this.START_TIME;
+    const balance = wei(await this.provider.getBalance(this.signer.address)).toNumber();
+    this.logger.info('Performing keeper healthcheck', { args: { uptime, balance } });
+
+    // A failure to submit metric should not cause application to halt. Instead, alerts will pick this up if it happens
+    // for a long enough duration. Essentially, do _not_ force keeper to slowdown operation just to track metrics
+    // for offline usage/monitoring.
+    this.metrics.time(Metric.KEEPER_UPTIME, uptime);
```

Comment:
Next TODO: keep track of metrics promises and timeout (or race them) to prevent accumulating unresolved metrics